verify_test.go

```diff
@@ -30,3 +30,56 @@ func TestVerifyNonce(t *testing.T) {
 		t.Errorf("verifyNonce failed unexpectedly: %v", err)
 	}
 }
+
+func TestVerifySignedFields(t *testing.T) {
+	v := url.Values{}
+
+	// Everything properly signed
+	v.Set("openid.signed", "signed,op_endpoint,claimed_id,identity,return_to,response_nonce,assoc_handle")
```

Comment:
Not a big deal, but it would be nicer if all those tests were in a table drive test (https://github.com/golang/go/wiki/TableDrivenTests)

Each test would have a list of key/values pairs for url.Values, as well as if an error is expected or not.